source/concepts/cryptography.rst

```diff
@@ -2,97 +2,78 @@
 Cryptography
 ############
 
-|codename| uses **elliptic curve cryptography** to verify the data integrity and to authenticate the signer's identity.
+|codename| uses `elliptic curve cryptography <https://en.wikipedia.org/wiki/Elliptic-curve_cryptography>`__ to verify data integrity and to authenticate a signer's identity.
 
 .. _keypair:
 
 ********
 Key pair
 ********
 
-Elliptic curve cryptography is an approach to public key cryptography.
-The cryptographic system uses **pairs of keys**:
+Elliptic curve cryptography is based on **key pairs**: a private and a matching public key. In particular, |codename| uses the `Twisted Edwards curve <https://en.wikipedia.org/wiki/Twisted_Edwards_curve>`__ with the digital signature algorithm named `Ed25519 <https://ed25519.cr.yp.to>`__ and hashing algorithm **SHA-512**:
 
-* **Private key**: A random 256-bit integer used to sign :ref:`entities <verifiable-entity>` known by the owner.
+* **Private key**: A random 256-bit (64 byte) integer used to sign :ref:`entities <verifiable-entity>` known by the owner.
 
-* **Public key**: The public identifier of the key pair, which can be disseminated widely. It is used to prove that the entity was signed with the paired private key.
+* **Public key**: A 128-bit (32 bytes) integer derived from the private key. It serves as the public identifier of the key pair and can be disseminated widely. It is used to prove that an entity was signed with the paired private key.
 
-The public key is cryptographically derived from the private key.
-In particular, |codename| uses the |edwards| with the digital signature algorithm named |Ed25519| and hashing algorithm **SHA-512**.
+The public key can be derived from the private key, but **not the other way around**.
 
-You can find the |implementation-derivation| under the ``crypto`` module of :doc:`catapult-server <../server>`.
+The implementation can be found in the `crypto module <https://github.com/nemtech/catapult-server/blob/main/src/catapult/crypto>`__  of :doc:`catapult-server <../server>`.
 
-*********
-Signature
-*********
-
-With a private key, the algorithm can sign messages producing 64-byte signatures.
-A signature is used to validate that a given key pair signed an entity just having the public key.
-
-You can find the implementation to |implementation-signature| and |implementation-verification| under the ``crypto`` module of :doc:`catapult-server <../server>`.
-
-.. _address:
-
-*******
-Address
-*******
+|codename| keys
+===============
 
-Public keys can be shared in a shorter form as addresses.
-A |codename| address is a **Base32 encoded triplet** consisting of:
+Key pairs are used in |codename| in different places, for **different purposes**. This is a summary of the keys used:
 
-* The network byte.
-* The 160-bit hash of the account's public key.
-* The 3 byte checksum, to allow the quick recognition of mistyped addresses.
+* **Main**: This key pair manages a **regular** :doc:`account`, containing mosaics.
```

Comment:
mosaics => bit limited since it also gives ownership of other assets, like namespace,s mosaics etc

Reply:
👍